hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/common/AbstractEventListenersAwareView.java

```diff
@@ -71,10 +82,24 @@ public void enter(final ViewChangeEvent event) {
         if (initial) {
             restoreState();
             initial = false;
-            return;
+        } else {
+            updateLayoutsOnViewEnter();
         }
 
-        updateLayoutsOnViewEnter();
+        if (StringUtils.hasText(event.getParameters())) {
+            handleStateParams(parseStateParameters(event.getParameters()));
+        }
+    }
+
+    private static Map<String, String> parseStateParameters(final String urlParams) {
+        return Arrays.stream(urlParams.split(DEFAULT_STATE_PARAMETER_SEPARATOR)).map(paramPair -> {
+            final String[] keyValue = paramPair.split(DEFAULT_STATE_PARAMETER_KEY_VALUE_SEPARATOR);
```

Comment:
[Optional] We could further safeguard us against invalid params
`paramPair.split(DEFAULT_STATE_PARAMETER_KEY_VALUE_SEPARATOR, 2);`
or alternatively in line 97
`if (keyValue.length >= 2) {`